### PR TITLE
Return error when can't get conn to send ACK

### PR DESCRIPTION
**Pull request by billfort**
Signed-off-by: bill fort <fxbao@hotmail.com>

### CODE REVIEW

File: session.go [hunk 1]
```diff
@@ -296,6 +297,8 @@ func (session *Session) ReceiveWith(localClientID, remoteClientID string, buf []
 
 		if conn, ok := session.connections[connKey(localClientID, remoteClientID)]; ok {
 			conn.SendAck(packet.SequenceId)
+		} else {
```
Comment (yilunzhang) [inline session.go:299]: I would recommend this flow:
if !ok return err
return conn.SendAck
This way we also handle and return the sendack error

Comment (billfort) [reply session.go:299]: Yes. I just checked, both heap.Push and conn.SendAck have no return value

File: session.go [hunk 2]
```diff
@@ -296,6 +297,8 @@ func (session *Session) ReceiveWith(localClientID, remoteClientID string, buf []
...
+			return fmt.Errorf("can't find conn %v to send ACK", connKey(localClientID, remoteClientID))
```
Comment (yilunzhang) [inline session.go]: Better to create a new error type so one can check type outside. It's probably more important than printing the client id
```
ConnNotFoundError  = NewGenericError("connection not found", false, false)
```

Comment (billfort) [reply session.go]: Yes, new error was created.

**Review by yilunzhang — APPROVED** [review]

**Pull request merged**